src/topology/sheaves/presheaf.lean

```diff
@@ -231,17 +230,85 @@ begin
     erw h (opens.op_map_id_obj U), simpa },
   { intros, apply pushforward.id_eq },
 end
-variables [has_colimits C]
+
+section iso
+
+/-- A homeomorphism of spaces gives an equivalence of categories of presheaves. -/
+@[simps] def presheaf_equiv_of_iso {X Y : Top} (H : X ≅ Y) :
+  X.presheaf C ≌ Y.presheaf C :=
+equivalence.congr_left (opens.map_map_iso H).symm.op
+
+variable {C}
+
+/--
+If `H : X ≅ Y` is a homeomorphism,
+then given an `H _* ℱ ⟶ 𝒢`, we may obtain an `ℱ ⟶ H ⁻¹ _* 𝒢`.
+-/
+def to_pushforward_of_iso {X Y : Top} (H : X ≅ Y) {ℱ : X.presheaf C} {𝒢 : Y.presheaf C}
+  (α : H.hom _* ℱ ⟶ 𝒢) : ℱ ⟶ H.inv _* 𝒢 :=
+(presheaf_equiv_of_iso _ H).to_adjunction.hom_equiv ℱ 𝒢 α
+
+@[simp]
+lemma to_pushforward_of_iso_app {X Y : Top} (H₁ : X ≅ Y) {ℱ : X.presheaf C} {𝒢 : Y.presheaf C}
+  (H₂ : H₁.hom _* ℱ ⟶ 𝒢) (U : (opens X)ᵒᵖ) :
+(to_pushforward_of_iso H₁ H₂).app U =
+  ℱ.map (eq_to_hom (by simp [opens.map, set.preimage_preimage])) ≫
+  H₂.app (op ((opens.map H₁.inv).obj (unop U))) :=
+begin
+  delta to_pushforward_of_iso,
+  simp only [equiv.to_fun_as_coe, nat_trans.comp_app, equivalence.equivalence_mk'_unit,
+    eq_to_hom_map, presheaf_equiv_of_iso_unit_iso_hom_app_app, equivalence.to_adjunction,
+    equivalence.equivalence_mk'_counit, presheaf_equiv_of_iso_inverse_map_app,
+    adjunction.mk_of_unit_counit_hom_equiv_apply],
+  congr
+end
+
+/--
+If `H : X ≅ Y` is a homeomorphism,
+then given an `H _* ℱ ⟶ 𝒢`, we may obtain an `ℱ ⟶ H ⁻¹ _* 𝒢`.
+-/
+def pushforward_to_of_iso {X Y : Top} (H₁ : X ≅ Y) {ℱ : Y.presheaf C} {𝒢 : X.presheaf C}
+  (H₂ : ℱ ⟶ H₁.hom _* 𝒢) : H₁.inv _* ℱ ⟶ 𝒢 :=
+((presheaf_equiv_of_iso _ H₁.symm).to_adjunction.hom_equiv ℱ 𝒢).symm H₂
+
+@[simp]
+lemma pushforward_to_of_iso_app {X Y : Top} (H₁ : X ≅ Y) {ℱ : Y.presheaf C} {𝒢 : X.presheaf C}
+  (H₂ : ℱ ⟶ H₁.hom _* 𝒢) (U : (opens X)ᵒᵖ) :
+(pushforward_to_of_iso H₁ H₂).app U =
+  H₂.app (op ((opens.map H₁.inv).obj (unop U))) ≫
+  𝒢.map (eq_to_hom (by simp [opens.map, set.preimage_preimage])) :=
+by simpa [pushforward_to_of_iso, equivalence.to_adjunction]
+
+end iso
+
+variables (C) [has_colimits C]
 
 /-- Pullback a presheaf on `Y` along a continuous map `f : X ⟶ Y`, obtaining a presheaf
 on `X`. -/
-@[simps]
+@[simps map_app]
 def pullback {X Y : Top.{v}} (f : X ⟶ Y) : Y.presheaf C ⥤ X.presheaf C := Lan (opens.map f).op
 
+@[simp] lemma pullback_obj_eq_pullback_obj {C} [category C] [has_colimits C] {X Y : Top.{v}}
+  (f : X ⟶ Y) (ℱ : Y.presheaf C) : (pullback C f).obj ℱ = pullback_obj f ℱ := rfl
+
 /-- The pullback and pushforward along a continuous map are adjoint to each other. -/
 @[simps unit_app_app counit_app_app]
 def pushforward_pullback_adjunction {X Y : Top.{v}} (f : X ⟶ Y) :
   pullback C f ⊣ pushforward C f := Lan.adjunction _ _
 
+/-- Pulling back along a homeomorphism is the same as pushing forward along its inverse. -/
+def pullback_hom_iso_pushforward_inv {X Y : Top.{v}} (H : X ≅ Y) :
```

Comment:
I think there should be simp-lemmas explaining how this definition unfolds when it is applied.

Reply:
After fiddling with it for a bit, I don't really have any suitable simp lemmas to add without an application at hand. Unfolding it straightforwardly gives an `colimit.desc` and `colimit.ι` that does not seem fun to work with.

Reply:
Hmmz, that's what I feared. But it means that users should never unfold this, and treat it as a black box. But now it is a black box with no lemmas about it. So it is basically just any random isomorphism between two functors. There should be some way to express that this is the "canonical" isomorphism.
At least, if `H` is `iso.refl _` then this isomorphism should be some sort of identity. And the construction is compatible with transitive isos. That's not yet pinning it down uniquely, but they still seem useful coherence lemmas to me.

Reply:
I think the best I could come up with is some sort of compatibility with the unit/counit.
This should probably be stated in terms of `adjunction.left_adjoint_uniq`, and specialized here.

Reply:
Would you mind trying that out?

Reply:
#10551 

Should I make this an abbreviation so that the simp lemmas can apply, or should I specialize them by hand?